Modules/_zstd/compressor.c

```diff
@@ -38,6 +39,9 @@ typedef struct {
 
     /* Compression level */
     int compression_level;
+
+    /* Lock to protect the compression context */
+    PyMutex lock;
```

Comment:
I think this is a right idea. We need a lock for both the GIL-enabled and FT builds, because the GIL doesn't protect things within a `Py_BEGIN_ALLOW_THREADS`, which is used before many ZSTD API calls.